File: packages/emitter/testcases/types/tuples-intersections/TuplesAndIntersections.ts
// Tuple types
export type Point2D = [number, number];
export type Point3D = [number, number, number];
export type NamedPoint = [x: number, y: number];

// Variadic tuples
export type Coords = [...number[]];
export type StringWithNumbers = [string, ...number[]];

// Intersection types
export interface Named {
  name: string;
}

export interface Aged {
  age: number;
}

export type Person = Named & Aged;

// Complex intersection
export interface Serializable {
  toJSON(): string;
}

export type SerializablePerson = Person & Serializable;

// Functions using tuples
export function distance(point: Point2D): number {
  const [x, y] = point;
  return Math.sqrt(x * x + y * y);
}

export function createPoint(x: number, y: number): Point2D {
  return [x, y];
}

// Function using intersection
export function greetPerson(person: Person): string {
  return `${person.name} is ${person.age} years old`;
}

// Rest elements with tuples
export function sum(...nums: [...number[]]): number {
  return nums.reduce((a, b) => a + b, 0);
}

File: packages/emitter/testcases/types/tuples-intersections/TuplesAndIntersections.cs
namespace TestCases.types.tuplesintersections
{
    public class Named
    {
        public string name { get; set; }
    }
    public class Aged
    {
        public double age { get; set; }
    }
    public class Serializable
    {
        public string toJSON() => throw new global::System.NotImplementedException();
    }

            public static class TuplesAndIntersections
            {
                // type Point2D = dynamic

                // type Point3D = dynamic

                // type NamedPoint = dynamic

                // type Coords = dynamic

                // type StringWithNumbers = dynamic

                // type Person = object

                // type SerializablePerson = object

                public static double distance(Point2D point)
                    {
                    var x = global::Tsonic.Runtime.Array.get(point, 0.0);
                    var y = global::Tsonic.Runtime.Array.get(point, 1.0);
                    return global::Tsonic.JSRuntime.Math.sqrt(x * x + y * y);
                    }

                public static Point2D createPoint(double x, double y)
                    {
                    return new global::System.Collections.Generic.List<object> { x, y };
                    }

                public static string greetPerson(Person person)
                    {
                    return $"{person.name} is {person.age} years old";
                    }

                public static double sum(dynamic nums)
                    {
                    return global::Tsonic.JSRuntime.Array.reduce(nums, (a, b) => a + b, 0);
                    }
            }
}
